test/39-compose-images.spec.ts

```diff
@@ -0,0 +1,530 @@
+				],
+				SharedSize: -1,
+				Size: 217024558,
+				VirtualSize: 217024558,
+			},
+			'sha256:25d6abae14f08de6b80f9d95003e674598738959a535a2f21be34c03675ebd02': {
+				Containers: -1,
+				Created: 1599146808,
+				Id:
+					'sha256:25d6abae14f08de6b80f9d95003e674598738959a535a2f21be34c03675ebd02',
+				Labels: { 'io.balena.qemu.version': '4.0.0+balena-aarch64' },
+				ParentId: '',
+				RepoDigests: [
+					'balena/aarch64-supervisor@sha256:e7e3b166e855f4c113a67bc528b4ef77408d05e280052f35452f3e2cd7b5322b',
+				],
+				RepoTags: ['balena/aarch64-supervisor:v11.14.0'],
+				SharedSize: -1,
+				Size: 72305850,
+				VirtualSize: 72305850,
+			},
+			'sha256:60783b8688d395f9b4ce4b288d941244d1b0a4c43114ba980acd012ccffc6b53': {
+				Containers: -1,
+				Created: 1585746557,
+				Id:
+					'sha256:60783b8688d395f9b4ce4b288d941244d1b0a4c43114ba980acd012ccffc6b53',
+				Labels: { 'io.balena.qemu.version': '4.0.0+balena-aarch64' },
+				ParentId: '',
+				RepoDigests: [
+					'balenalib/aarch64-alpine-supervisor-base@sha256:6eb712fc797ff68f258d9032cf292c266cb9bd8be4cbdaaafeb5a8824bb104fd',
+				],
+				RepoTags: ['balenalib/aarch64-alpine-supervisor-base:3.11'],
+				SharedSize: -1,
+				Size: 14503741,
+				VirtualSize: 14503741,
+			},
+			'sha256:a29f45ccde2ac0bde957b1277b1501f471960c8ca49f1588c6c885941640ae60': {
+				Containers: -1,
+				Created: 1578015959,
+				Id:
+					'sha256:a29f45ccde2ac0bde957b1277b1501f471960c8ca49f1588c6c885941640ae60',
+				Labels: null,
+				ParentId: '',
+				RepoDigests: null,
+				RepoTags: ['balena-healthcheck-image:latest'],
+				SharedSize: -1,
+				Size: 9136,
+				VirtualSize: 9136,
+			},
+		};
+		// Stub the database to return images we want
+		mockedDatabase.setImages(IMAGES_FROM_DB).stub();
+		// Perform the test with our specially crafted data
+		await mockedDockerode.testWithData({ images }, async () => {
+			// Check that a single image is returned when given entire object
+			expect(
+				db.models('image').select().where(IMAGE_TO_REMOVE),
+			).to.eventually.have.lengthOf(1);
+			// Check that only one image with this dockerImageId exists in the db
+			expect(
+				db
+					.models('image')
+					.where({ dockerImageId: IMAGE_TO_REMOVE.dockerImageId })
+					.select(),
+			).to.eventually.have.lengthOf(1);
+			// Now remove this image...
+			await Images.remove(IMAGE_TO_REMOVE);
+			// Check that docker does not have this image
+			await expect(docker.getImage(IMAGE_TO_REMOVE.name).inspect()).to
+				.eventually.be.empty;
+			// Check that the database longer has this image
+			await expect(db.models('image').select().where(IMAGE_TO_REMOVE)).to
+				.eventually.be.empty;
+			// Check that docker remove was called once
+			const removeSteps = _(mockedDockerode.actions)
+				.pickBy({ name: 'remove' })
+				.map()
+				.value();
+			expect(removeSteps).to.have.lengthOf(1);
+		});
+	});
+
+	it('Removes an Image with digests', async () => {
```

Comment:
This uses real data that causes this bug to actually happen. We are adding this test coverage precisely because it caused us to write a new PR to fix that bug and this is the exact payloads that will cause it.